collector/memory_bsd.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build freebsd dragonfly
+// +build !nomeminfo
+
+package collector
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	memorySubsystem = "memory"
+)
+
+type memoryCollector struct {
+	pageSize uint64
+	sysctls  []bsdSysctl
+	kvm      kvm
+}
+
+func init() {
+	registerCollector("meminfo", defaultEnabled, NewMemoryCollector)
+}
+
+// NewMemoryCollector returns a new Collector exposing memory stats.
+func NewMemoryCollector() (Collector, error) {
+	tmp32, err := unix.SysctlUint32("vm.stats.vm.v_page_size")
+	if err != nil {
+		return nil, fmt.Errorf("sysctl(vm.stats.vm.v_page_size) failed: %s", err)
+	}
+	size := float64(tmp32)
+
+	fromPage := func(v float64) float64 {
+		return v * size
+	}
+
+	return &memoryCollector{
+		pageSize: uint64(tmp32),
+		sysctls: []bsdSysctl{
+			// Descriptions via: https://wiki.freebsd.org/Memory
+			{
+				name:        "active_bytes",
+				description: "Recently used by userland",
+				mib:         "vm.stats.vm.v_active_count",
+				conversion:  fromPage,
+			},
+			{
+				name:        "inactive_bytes",
+				description: "Not recently used by userland",
+				mib:         "vm.stats.vm.v_inactive_count",
+				conversion:  fromPage,
+			},
+			{
+				name:        "wired_bytes",
+				description: "Locked in memory by kernel, mlock, etc",
+				mib:         "vm.stats.vm.v_wire_count",
+				conversion:  fromPage,
+			},
+			{
+				name:        "cache_bytes",
+				description: "Almost free, backed by swap or files, available for re-allocation",
+				mib:         "vm.stats.vm.v_cache_count",
+				conversion:  fromPage,
+			},
+			{
+				name:        "buffer_bytes",
+				description: "Disk IO Cache entries for non ZFS filesystems, only usable by kernel",
+				mib:         "vfs.bufspace",
```

Comment:
Proposal : 
`dataType:    bsdSysctlTypeUint64,`

Reply:
Thanks for the heads up.  Will rebase on master when #710 is solved.